Lib/fontTools/pens/pointPen.py

```diff
@@ -260,27 +271,35 @@ def moveTo(self, pt):
 		self.contour.append((pt, "move"))
 
 	def lineTo(self, pt):
-		assert self.contour is not None, "contour missing required initial moveTo"
+		if self.contour is None:
+			raise PenError("Contour missing required initial moveTo")
 		self.contour.append((pt, "line"))
 
 	def curveTo(self, *pts):
-		assert self.contour is not None, "contour missing required initial moveTo"
+		if not pts:
```

Comment:
Should this be TypeError? Likewise qCurveTo.

Other than that: LGTM, thanks!

Reply:
Hm.

https://docs.python.org/3/library/exceptions.html?highlight=typeerror#TypeError

> Passing arguments of the wrong type (e.g. passing a list when an int is expected) should result in a TypeError, but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a ValueError.

I'll change it to a ValueError but then I need to change my Hypothesis thing :thinking:

Reply:
What you quote doesn't seem relevant for the case here.

```python
>>> def foo(a, b, c):
...     ...
... 
>>> foo()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: foo() missing 3 required positional arguments: 'a', 'b', and 'c'
>>> 
```

Reply:
Or:
```python
>>> def foo(pt1, *pts):
...     ...
... 
>>> pts = ()
>>> foo(*pts)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: foo() missing 1 required positional argument: 'pt1'
>>> 
```

Reply:
🤔 🤔 🤔 

```text
In [3]: class A:
   ...:     def curveTo(self, *pts):
   ...:         print(self, repr(pts))
   ...:

In [4]: a = A()

In [5]: a.curveTo(1,2,3)
<__main__.A object at 0x00000245B88029A0> (1, 2, 3)

In [6]: a.curveTo()
<__main__.A object at 0x00000245B88029A0> ()

In [7]: a.curveTo(())
<__main__.A object at 0x00000245B88029A0> ((),)

In [8]: pts = ()
In [9]: a.curveTo(*pts)
<__main__.A object at 0x00000245B88029A0> ()
```

We always get an empty sequence as a fallback, so ValueError seems relevent to me?

Reply:
`*pts` represents an arbitrary number of arguments. You want to ensure the method called with at least one argument. Calling a function with the wrong number of argument ordinarily raises TypeError, hence my suggestion.

Reply:
Oh, I see what you're getting at now. TypeError it is then!